lib/events.js

```diff
@@ -485,3 +486,26 @@ function unwrapListeners(arr) {
   }
   return ret;
 }
+
+function once(emitter, name) {
+  return new Promise((resolve, reject) => {
+    const mainListener = (...args) => {
+      if (secondListener !== undefined) {
+        emitter.removeListener('error', secondListener);
+      }
+      resolve(args);
+    };
+    let secondListener;
```

Comment:
This is just personal preference but having this at the start of the function body would be clearer for me. When I read the code, the secondListener is referenced before its defined which makes it somewhat confused. I know it's kind of nitpicky tho so non-blocking... 

(Maybe naming it `errorRejectListener` would also be clearer?)

Reply:
`eventListener` and `errorListener` (or something of the like) would be clearer than `main` and `second`.